### ISSUE TITLE
License: Clarifying facts about the EUPL

### ISSUE THREAD

**Issue by hastebrot** [TestFX/TestFX#177]
Here is my try to give a quick overview for the EUPL. My source are the "EUPL Guidelines for users and developers", which are good to understand even for the layman.

---
### License

TestFX is “Free and Open Source software” (F/OSS) and licensed under the “European Union Public Licence” (EUPL).

Here are some quick facts for a first glance to make the license easier to understand:
- You receive a series of world-wide, permanent, royalty-free, and non-exclusive rights. This means, for example, you can use TestFX in Europe and outside of Europe, and the rights are given forever.
- You can use TestFX in any circumstance and for all usage. This means, for example, that both private and public uses, non-profit or for-profit uses are allowed. Your program distributed along with it is not restricted to be open-source software. You can use TestFX within a single application using static or dynamic linking, but without merging the source codes.
- If you make modifications and decide to re-distribute the derivative work, the forked derivative must be under the EUPL. If you incorporate pieces of software code licensed under a “permissive“ license (e.g. MIT or BSD), you re-distribute the combined derivative under the provision of the EUPL. If you incorporate pieces of software code licensed under a “copyleft“ license (e.g. GPL or EPL), you re-redistribute the combined derivative under the provision of the “copyleft“ license.

For the full license text and more detailed explanations (available in 22 languages):
- Text and preamble of the European Union Public Licence (EUPL)<br>
  https://joinup.ec.europa.eu/software/page/eupl/licence-eupl
- EUPL Guidelines for users and developers<br>
  https://joinup.ec.europa.eu/software/page/eupl/eupl-guidelines

The license header:

```
Copyright 2013-2014 SmartBear Software
Copyright 2014-2015 The TestFX contributors

Licensed under the EUPL, Version 1.1 or - as soon they will be approved by the European
Commission - subsequent versions of the EUPL (the "Licence"); You may not use this work
except in compliance with the Licence.

You may obtain a copy of the Licence at:
http://ec.europa.eu/idabc/eupl.html

Unless required by applicable law or agreed to in writing, software distributed under 
the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF 
ANY KIND, either express or implied. See the Licence for the specific language 
governing permissions and limitations under the Licence.
```

**Comment by puce77** [comment]
As I understand this license, TestFX cannot be used in libraries covered by a weak copyleft license such as CDDL or in commercial closed source projects?

**Comment by brcolow** [comment]
"This means, for example, that both private and public uses, non-profit or for-profit uses are allowed. Your program distributed along with it is not restricted to be open-source software"

I would say that directly contradicts your understanding.

**Comment by hastebrot** [comment]
Hi! TestFX can definately be used in commercial projects. Actually the EUPL is very commercial-friendly.

**Comment by hastebrot** [comment]
CDDL: The question is, can you distribute you code that uses TestFX under the CDDL, or to you have to distribute it under the EUPL. I'm not sure what applies here with a file-based copyleft license, and if we merge codebases or just link to a library.

**Comment by hastebrot** [comment]
Here are two documents that discuss compatibility to other licenses (like CDDL):

- https://joinup.ec.europa.eu/software/page/eupl/eupl-compatible-open-source-licences
- https://joinup.ec.europa.eu/system/files/doc/Doc3ef5.pdf (use `wget`, if you browser tries to show this as `text/html`)

**Comment by puce77** [comment]
From what I understand: You can distribute an application under EUPL under some circumstances, even if not all parts are published under EUPL (upstream). But you cannot distribute a library which is under EUPL together with an application, which is not covered by a license in the EUPL compatibility list (downstream). And that's my point: you cannot distribute it as part of an application which uses a weak copy left license (such as CDDL), a permissive license (such as Apache License) or is a closed sourced application. Is this correct?

**Comment by puce77** [comment]
@brcolow Yes, I've seen this in the initial comment by @hastebrot. But where does this statement come from?  It seems to contradict the downstream and upstream definition in https://joinup.ec.europa.eu/software/page/eupl/eupl-compatible-open-source-licences

**Comment by hastebrot** [comment]
@puce77 This is my interpretation of the EUPL guide

>Here is my try to give a quick overview for the EUPL. My source are the "EUPL Guidelines for users and developers"

**Comment by puce77** [comment]
And another question is how EUPL affects applications, when the EUPL covered library (eg. TestFX) is only used in the test code but not part of the distributed application.
